src/main/java/net/dv8tion/jda/api/JDA.java

```diff
@@ -1117,6 +1117,60 @@ default User getUserByTag(@Nonnull String username, @Nonnull String discriminato
     @CheckReturnValue
     RestAction<User> retrieveUserById(long id, boolean update);
 
+    /**
+     * Retrieves a {@link Channel} object based on the provided id.
+     *
+     * <p>The returned {@link net.dv8tion.jda.api.requests.RestAction RestAction} can encounter the following Discord errors:
+     * <ul>
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_CHANNEL ErrorResponse.UNKNOWN_CHANNEL}
+     *     <br>Occurs when the provided id does not refer to a {@link Channel Channel} known by Discord.
+     *     known by Discord. Typically occurs when developers provide an incomplete id (cut short).</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_ACCESS MISSING_ACCESS}
+     *     <br>Occurs when the provided id does refers to a {@link Channel Channel} that this account does not have access to.
+     *     </li>
+     * </ul>
+     *
+     * @param  id
+     *         The id of the requested {@link Channel Channel}.
+     *
+     * @return {@link net.dv8tion.jda.api.requests.RestAction RestAction} - Type: {@link net.dv8tion.jda.api.entities.Channel Channel}
+     *         <br>On request, gets the User with id matching provided id from Discord.
+     */
+    @Nonnull
+    @CheckReturnValue
+    RestAction<Channel> retrieveChannelById(long id);
```

Comment:
We might not actually want to publicly expose this yet. If a developer attempts to retrieve a channel that is part of a guild but that guild isn't loaded in this shard, then the EntityBuilder would likely explode trying to craft the channel.

We probably wanna hold off on public access to this functionality until we work on the RestOnly Module.